### PR TITLE
Friends Endpoints

### LINKED ISSUE #149: Friends Component Endpoints
The issue to keep track of adding the Friends component endpoints:

- `create_item` - create friendship
- `delete_item` - delete friendship
- `update_item` - accept/reject friendship request
- `check_item` - check friendship
- `get_items` - List of user's friends.

### PR DESCRIPTION
**Pull request by renatonascalves**
closes #149

### CODE REVIEW

File: includes/bp-friends/classes/class-bp-rest-friends-endpoint.php [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+
+	/**
+	 * Delete a friendship.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function delete_item( $request ) {
```
Comment (imath) [inline includes/bp-friends/classes/class-bp-rest-friends-endpoint.php]: Where is the corresponding route in BP_REST_Friends_Endpoint::register_routes() ?

File: includes/bp-friends/classes/class-bp-rest-friends-endpoint.php [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				),
+				'schema' => array( $this, 'get_item_schema' ),
+			)
+		);
```
Comment (imath) [inline includes/bp-friends/classes/class-bp-rest-friends-endpoint.php]: Looks like `friendship/id` routes are missing ?
-> get_item() ?
-> update_item() ?
-> delete_item() ?

**Comment by renatonascalves** [comment]
@imath Pushed just for visibility. I'm aware of the other endpoints/methods. I'm working on that. :)

File: includes/bp-friends/classes/class-bp-rest-friends-endpoint.php (hunk 1 above)
Comment (renatonascalves) [reply includes/bp-friends/classes/class-bp-rest-friends-endpoint.php]: @imath Didn't understand the question... :/

Comment (renatonascalves) [reply includes/bp-friends/classes/class-bp-rest-friends-endpoint.php]: Ah, I see it! I haven't added it yet. I will though. :)

File: includes/bp-friends/classes/class-bp-rest-friends-endpoint.php (hunk 2 above)
Comment (renatonascalves) [reply includes/bp-friends/classes/class-bp-rest-friends-endpoint.php]: Correct! I'm working on that! :)

**Comment by imath** [comment]
Awesome, well then I'd say if you want to commit a first pass, I'd suggest to remove the delete_item() method or to add the corresponding route :)

**Comment by renatonascalves** [comment]
@imath Now I think I'm done with the endpoint.

Could you take another look?

**Pull request merged**